File: resharper/test/data/References/InvalidPropertyDataProperties.vb
Imports System
Imports System.Collections.Generic
Imports Xunit
Imports Xunit.Extensions

Public Class MyTest
    <Theory>
    <PropertyData("NonPublicDataEnumerator")>
    Public Sub Test1(ByVal value As Int32)
        Assert.Equal(42, value)
    End Sub

    <Theory>
    <PropertyData("NonStaticDataEnumerator")>
    Public Sub Test1(ByVal value As Int32)
        Assert.Equal(42, value)
    End Sub

    <Theory>
    <PropertyData("InvalidReturnTypeDataEnumerator")>
    Public Sub Test1(ByVal value As Int32)
        Assert.Equal(42, value)
    End Sub

    Private Shared ReadOnly Iterator Property NonPublicDataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property

    Public ReadOnly Iterator Property NonStaticDataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property

    Public Shared ReadOnly Iterator Property NonStaticDataEnumerator() As IEnumerable(Of Int32())
        Get
            Yield New Int32() { 42 }
        End Get
    End Property
End Class

File: resharper/test/data/References/InvalidPropertyDataProperties.cs
using System.Collections.Generic;
using Xunit;
using Xunit.Extensions;

namespace Foo
{
    public class MyTest
    {
        [Theory]
        [PropertyData("NonPublicDataEnumerator")]
        public void Test1(int value)
        {
            Assert.Equal(42, value);
        }

        [Theory]
        [PropertyData("NonStaticDataEnumerator")]
        public void Test2(int value)
        {
            Assert.Equal(42, value);
        }

        [Theory]
        [PropertyData("InvalidReturnTypeDataEnumerator")]
        public void Test3(int value)
        {
            Assert.Equal(42, value);
        }

        // Needs to be public
        private static IEnumerable<object[]> NonPublicDataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }

        // Needs to be static
        public IEnumerable<object[]> NonStaticDataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }

        // Invalid return type signature
        public static IEnumerable<int[]> InvalidReturnTypeDataEnumerator
        {
            get { return new List<int[]>(new int[] { 42 }); }
        }
    }
}
